packages/react-form/src/hooks/list/hooks/reducer.ts

```diff
@@ -158,6 +175,20 @@ function reduceList<Item extends object>(
         list: action.payload.list.map(initialListItemState),
       };
     }
+    case 'moveFieldItem': {
+      const newList = [...state.list];
+      newList.splice(action.payload.oldIndex, 1);
+      newList.splice(
+        action.payload.newIndex,
+        0,
+        action.payload.item as FieldStates<Item>,
```

Comment:
I believe if we just grab the item from the list, there might be no need to typecast to FieldStates<Item> when adding it to the newIndex

Reply:
👍